projects/dex-ui/src/components/Liquidity/LiquidityRoot.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import { Well } from "@beanstalk/sdk/Wells";
+import { AddLiquidity } from "./AddLiquidity";
+import { RemoveLiquidity } from "./RemoveLiquidity";
+import SlippagePanel from "./SlippagePanel";
+
+type LiquidityRootProps = {
+  well: Well;
+  txnCompleteCallback: () => void;
+};
+
+export const LiquidityRoot = ({ well, txnCompleteCallback }: LiquidityRootProps) => {
+  const [showRemove, setShowRemove] = useState<boolean>(false);
+
+  // Slippage-related
+  const [showSlippageSettings, setShowSlippageSettings] = useState<boolean>(false);
+  const [slippage, setSlippage] = useState<number>(0.1);
+
+  const slippageSettingsClickHandler = () => {
+    setShowSlippageSettings(true);
+  };
+
+  const slippageSettingsCloseHandler = () => {
+    setShowSlippageSettings(false);
+  };
```

Comment:
these should be useCallbacks since they're passed as props. also maybe combine into one `toggleSlippage()` that just flips the current state. 

Or do it inline like you did for the TabButton onclick. If you do it inline, it will cause the component to re-render since it's a new function when parent re-renders